clients/src/test/java/org/apache/kafka/common/security/oauthbearer/OAuthBearerLoginModuleTest.java

```diff
@@ -305,4 +319,54 @@ public void login1Commit1Login2Abort2Login3Commit3Logout3() throws LoginExceptio
         assertEquals(1, privateCredentials.size());
         assertSame(tokens[2], privateCredentials.iterator().next());
     }
+
+    @Test
+    public void commitPopulatesExtensions() throws LoginException {
+        Subject subject = new Subject();
+
+        // Create callback handler
+        OAuthBearerToken[] tokens = new OAuthBearerToken[] {EasyMock.mock(OAuthBearerToken.class),
+                EasyMock.mock(OAuthBearerToken.class), EasyMock.mock(OAuthBearerToken.class)};
+        EasyMock.replay(tokens[0], tokens[1], tokens[2]); // expect nothing
+        TestCallbackHandler testTokenCallbackHandler = new TestCallbackHandler(tokens);
+
+        // Create login modules
+        OAuthBearerLoginModule loginModule1 = new OAuthBearerLoginModule();
+        loginModule1.initialize(subject, testTokenCallbackHandler, Collections.<String, Object>emptyMap(),
+                Collections.<String, Object>emptyMap());
+
+        loginModule1.login();
+        // Should populate public credentials with an empty map and not throw an exception
+        loginModule1.commit();
+        SaslExtensions extensions = subject.getPublicCredentials(SaslExtensions.class).iterator().next();
+        assertFalse(extensions.map().isEmpty());
+        assertEquals("true", extensions.map().get("test"));
+    }
+
+    /**
+     * 2.1.0 added customizable SASL extensions and a new callback type.
+     * Ensure that old, custom-written callbackHandlers that do not handle the callback work
+     */
+    @Test
+    public void commitDoesNotThrowOnUnsupportedExtensionsCallback() throws LoginException {
```

Comment:
This test becomes unnecessary after weaving SaslExtensions into the above tests as long as you include at least one null element in the array for each one.

Reply:
No, we'd need this. The null element won't test out this backwards-compatible behavior

Reply:
This test is checking the same thing that was checked via passing in RAISE_UNSUPPORTED_CB_EXCEPTION_FLAG (null) at line 133 and checking for EMPTY_EXTENSIONS at lines 191 and 200.  This test should be deleted.

Reply:
Sorry about my initial comment of "No, we'd need this. The null element won't test out this backwards-compatible behavior". I unfortunately commented prematurely before completely understanding your suggestion.

I acknowledge that this is verified in the above tests.
It's just that from what I've read from TDD books, the overall approach experts recommend is to rely on single, small tests that test concrete functionality. This way, when a problem occurs you immediately know what the cause is - `commitPopulatesExtensions` - oh, my extensions weren't populated.

Where as if you get an error in test `login1Commit1Login2Abort2Login3Commit3Logout3` you need to investigate the test well and figure out where the problem is. 
While such big tests are always useful, I believe a test suite comprised of more smaller tests is better.
Tests comprised of more methods serve as better documentation. You can then just read the method names and get a general feeling of what the tested subject should do.

Please share your thoughts on this

Reply:
Good point -- I agree.  Let's keep this test and eliminate the null value in `login1Commit1Login2Abort2Login3Commit3Logout3`.